src/main/resources/com/ncc/aif/restricted_aif.shacl

```diff
@@ -0,0 +1,159 @@
+    sh:maxCount 2 ;
+    sh:message "Only 2 contained justifications allowed for an edge" .
+
+# Turn off CompoundJustificationMinimum globally
+aida:CompoundJustificationMinimum sh:deactivated true .
+
+#########################
+# Each edge justification must be represented uniformly in AIF by
+# aida:CompoundJustification, even if only one span is provided
+# Edges are assumed to be relation and event arguments
+#------------------------
+# Define restriction where justification can only be compound justification
+aida:EdgeJustificationCompound
+    a sh:PropertyShape ;
+    sh:path aida:justifiedBy ;
+    sh:class aida:CompoundJustification ;
+    sh:message "Edge justification must be of type aida:CompoundJustification" .
+
+# Enforce edge justification restriction on event and relation arguments
+aida:EventArgumentShape
+    sh:property aida:EdgeJustificationCompound, aida:EdgeJustificationLimit .
+aida:RelationArgumentShape
+    sh:property aida:EdgeJustificationCompound, aida:EdgeJustificationLimit .
+
+#########################
+# Video must use aida:KeyFrameVideoJustification. Remove ShotVideoJustification
+#------------------------
+# Restrict normal Justification types globally
+aida:JustificationPropertyShape
+    sh:xone (
+        [sh:class aida:TextJustification]
+        [sh:class aida:ImageJustification]
+        [sh:class aida:AudioJustification]
+        [sh:class aida:KeyFrameVideoJustification]
+        [sh:class aida:CompoundJustification] ) .
+
+# Restrict CompoundJustificationTypes globally
+aida:CompoundJustificationTypes
+    sh:xone (
+        [sh:class aida:TextJustification]
+        [sh:class aida:ImageJustification]
+        [sh:class aida:AudioJustification]
+        [sh:class aida:KeyFrameVideoJustification] ) .
+
+#########################
+# Members of clusters are entity objects, relation objects, and event objects (not clusters)
+#------------------------
+aida:ClusterMembershipShape
+    sh:property [
+        a sh:PropertyShape ;
+        sh:path aida:clusterMember ;
+        sh:xone (
+            [sh:class aida:Entity]
+            [sh:class aida:Event]
+            [sh:class aida:Relation]
+        ) ;
+        sh:message "Cluster member type not allowed to be part of cluster"
+    ] .
+
+#########################
+# Entity, Relation, and Event object is required to be part of at least one cluster.
+# This is true even if there is nothing else in the cluster
+#------------------------
+aida:EntityShape
+    sh:sparql [
+        sh:message "Entity is not clustered" ;
+        sh:select """
+            PREFIX aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#>
+            SELECT $this
+            WHERE {
+                FILTER NOT EXISTS {
+                    ?membership a aida:ClusterMembership .
+                    ?membership aida:clusterMember $this .
+                }
+            }
+        """ ;
+    ] .
+
+aida:EventShape
```

Comment:
Same as the comment above? Is there a reason the name of this is not more expressive like the `aida:RelationMustBeClustered`? Should we call this `aida:EventShapeNotClustered`?